### ISSUE TITLE
design not responsive when i tried to use small device screen

### ISSUE THREAD

**Issue by flutter-f** [OpenFlutter/flutter_screenutil#306]
now im using 5.0.0+3 version and i set splitScreenMode: false.
i tried this screen on my device (1440 x 2560) and it worked correctly, but when i tried on device (480 x 800) error happened.
i set these codes:
Image.asset(image, height: 332.h) // for image
Text(title, size: 40.sp) // for title
Text(description, size: 16.sp) // for description.

so i think we have issue in (.sp) , can you please help me ?
thank you very much.
![kk](https://user-images.githubusercontent.com/58072084/147365780-3e277023-02f9-405c-b9a5-07a7a0fcb8ee.PNG)
![kkk](https://user-images.githubusercontent.com/58072084/147365782-5cb34f62-c4be-48ba-8da7-8011079c5b44.PNG)

**Comment by AhmedLSayed9** [comment]
The UI design above is (390×844) and on device size (480 x 800) we will get `1.23` scaleWidth and `0.94` scaleHeight.

The new way of using `.w` for text adaption won't work with cases like this `scaleHeight<scaleWidth & screenHeight<designHeight` and could cause overflow like this issue scenario.

so I think back to the previous `.sp` calculation "Text adaptation according to the smaller of width or height" would help.

**Comment by lizhuoyuan** [comment]
will fix in 5.0.1+3 today

**Comment by AhmedLSayed9** [comment]
Thanks @lizhuoyuan for your great work!

I see now why you've removed height in last update. I've seen #297 issue now but that keyboard font resizing isn't because of the `.sp` calculation method. That code isn't correct, it has duplicated `MaterialApp` and `Scaffold` inside `runApp` and even with `.w` font adaption the `resizeToAvoidBottomInset = false` isn't working correctly because that duplicated `MaterialApp` and `Scaffold`.

I've tested the same sample code after correcting the duplication, It works perfect with old `.sp` with both `resizeToAvoidBottomInset` set to true and false:

```
import 'package:flutter/material.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';

main() async {
  //runApp(
  //  MaterialApp(
  //    home: Scaffold(body: EntryPoint()),
  //  ),
  //);
  runApp(
    EntryPoint(),
  );
}

class EntryPoint extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return ScreenUtilInit(
        designSize: Size(1080, 1920),
        builder: () => MaterialApp(
          home: Scaffold(
            resizeToAvoidBottomInset: false,
            body: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                TextField(
                  style: TextStyle(fontSize: 180.sp),
                ),
              ],
            ),
          ),
        ));
  }
}
```

<p align="center">
    <img src="https://user-images.githubusercontent.com/70890146/147378679-0b61f80a-3f76-4597-9bf3-edea7781fc56.jpg" alt="Image" width="250"/>
    <img src="https://user-images.githubusercontent.com/70890146/147378697-587ce0a0-968f-4674-918b-68cf982b9827.jpg" alt="Image" width="250"/>
    <img src="https://user-images.githubusercontent.com/70890146/147378704-71aa6c4d-8379-4f4a-bb04-512ebc5f8fbf.jpg" alt="Image" width="250"/>
</p>

So, there's no need for adding two ways of Font adaption anyway.
